test/test-BlueprintParser.cc

```diff
@@ -816,8 +816,8 @@ TEST_CASE("Report error when a resource attributes type is circularly referenced
     ParseResult<Blueprint> blueprint;
     SectionParserHelper<Blueprint, BlueprintParser>::parse(source, BlueprintSectionType, blueprint, ExportSourcemapOption, Models(), &blueprint);
 
-    REQUIRE(blueprint.report.error.code == MSONError);
-    SourceMapHelper::check(blueprint.report.error.location, 65, 13);
+    REQUIRE(blueprint.report.error.code == Error::OK);
+    REQUIRE(blueprint.report.warnings.empty());
```

Comment:
Here we need to add `REQUIRE`s to check that base type of `Post` and `B` are `mson::ImplicitObjectBaseType`.